packages/common/src/TransactionUtils.ts

```diff
@@ -262,3 +307,29 @@ export async function findBlockNumberAtTimestamp(
   }
   return { blockNumber: block.number, error: Math.abs(targetTimestamp - parseInt(block.timestamp.toString())) };
 }
+
+/**
+ * @notice Consumes an array of transactions with embedded promises produced by iteratively calling runTransaction.
+ * Waits on all transactions to settle within the batch (included in a block). If any transaction contains an error then
+ * produce a log to that effect. This method is intended to be called at the end of a bot run cycle to ensure that all
+ * transactions that were submitted were indeed included without error. Note that runTransaction will not submit a
+ * transaction if the function can detect it will revert (i.e using the .call syntax). Therefore this function will only
+ * catch reverts that could not be seen at submission time.
+ */
+export async function processTransactionPromiseBatch(transactions: Array<executedTransaction>, logger: winston.Logger) {
+  logger.debug({
+    at: "TransactionUtils",
+    message: "Waiting on transaction batch",
+    transactions: transactions.map((transaction) => transaction.transactionHash),
+  });
+  const transactionResults = await Promise.allSettled(transactions.map((tx: any) => tx.receipt));
+  const revertedTransactions = (transactionResults as any).filter((result: any) => result.status === "rejected");
+  if (revertedTransactions.length == 0)
+    logger.debug({ at: "TransactionUtils", message: "Transaction batch processed without error" });
+  else
+    logger.error({
+      at: "TransactionUtils",
+      message: "Transaction batch processed with error",
+      errors: revertedTransactions.map((transaction: any) => transaction.reason),
```

Comment:
Would it be useful to include the transaction hashes for those that failed? It seems like we would get a previous log telling us the txn was sent, so it might be nice to be able to link each error back to its txn hash. Thoughts? Would this be difficult to add?

Reply:
this `reason` will include the transaction has. it'll look something like this:
```
{
 "blockHash": "0xd1c3f3557c5819cf59b0c8d833b03439fcf458d0c1831dce0dccd6965e32b1f2",
 "blockNumber": 13707214,
 "contractAddress": null,
 "cumulativeGasUsed": 12226892,
 "effectiveGasPrice": "0x1912a70f32",
 "from": "0x428ab2ba90eba0a4be7af34c9ac451ab061ac010",
 "gasUsed": 32907,
 "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
 "status": false,
 "to": "0x7355efc63ae731f584380a9838292c7046c1e433",
 "transactionHash": "0xf9039dcef419fc43d198b2bd6c18235f1c3b50ba0ed414a6e2b533bd9b4b8b27",
 "transactionIndex": 223,
 "type": "0x2",
 "events": {}
}
```

so we get the tx hash from this for free